### PR TITLE
Send `/{send,make}_knock` to Synapse workers

### PR DESCRIPTION
**Pull request by anoadragon453**
These endpoints should be handled by workers correctly.

To test this configuration, before it's recommended in https://github.com/element-hq/synapse/pull/17058.

### CODE REVIEW

File: lib/SyTest/Homeserver/Synapse.pm [hunk 1]
```diff
@@ -1257,6 +1257,8 @@ sub generate_haproxy_map
 ^/_matrix/federation/v1/make_leave/                   federation_reader
 ^/_matrix/federation/v1/send_join/                    federation_reader
 ^/_matrix/federation/v1/send_leave/                   federation_reader
+^/_matrix/federation/(v1|v2)/make_knock/              federation_reader
+^/_matrix/federation/(v1|v2)/send_knock/              federation_reader
```
Comment (clokep) [inline lib/SyTest/Homeserver/Synapse.pm:1261]: I think only a v1 version exists? (Same goes for complement and the docs btw.)

Comment (anoadragon453) [reply lib/SyTest/Homeserver/Synapse.pm:1261]: D'oh, good catch. Copy/paste fail.

Comment (clokep) [reply lib/SyTest/Homeserver/Synapse.pm]: Although shouldn't the join, leave, etc ones include v2? 🤔

Comment (anoadragon453) [reply lib/SyTest/Homeserver/Synapse.pm]: Another good catch! I've corrected in https://github.com/matrix-org/sytest/pull/1382.

**Review by erikjohnston — APPROVED** [review]

**Pull request merged**